### DISCUSSION TITLE
Engine RPC API not available

### DISCUSSION THREAD

**Discussion by yakimant** [erigontech/erigon#9599] [Q&A]
Hi guys,

I'm new to Erigon. Here is an example for `engine_newPayloadV2` RPC method:

```
{
  "jsonrpc": "2.0",
  "id": 1,
  "error": {
    "code": -32601,
    "message": "the method engine_newPayloadV2 does not exist/is not available"
  }
}
```

Although I can see it should be available:
https://github.com/ledgerwatch/erigon/tree/devel/cmd/rpcdaemon

I run with `--http.api=eth,net,erigon,admin,engine`

Thanks!

**Comment by AskAlexSharov** [comment]
1. maybe you mess up with `*` - it's special character in shell and can destroy flags parsing.
2. maybe you didn't configure `--authrpc`

**Reply by yakimant** [reply]
Just checked the full command - doesn't contain any `*` char.

- `--http.addr=0.0.0.0 --http.port=8545`
- `--http.api=eth,net,erigon,admin,engine`
- `--authrpc.addr=0.0.0.0 --authrpc.port=8551 --authrpc.jwtsecret=/keys/jwtsecret`

This is how I test RPC method:
```
curl -X POST --silent --max-time 5 --show-error --fail-with-body -H Content-type:application/json --data '{
  "id": 1,
  "jsonrpc": "2.0",
  "method": "engine_newPayloadV2",
  "params": []
}' http://0.0.0.0:8545/
```

**Reply by yakimant** [reply]
Do I understand correctly, that I should use `auth.rpc` port for all `engine` methods?
https://github.com/ethereum/execution-apis/blob/main/src/engine/authentication.md

**Comment by AskAlexSharov** [comment]
probably it's done already. closing
